dataflow/src/node/process.rs

```diff
@@ -202,6 +202,40 @@ impl Node {
                     _ => {}
                 };
 
+                // When we miss at a can_query_through node we have to replay from
+                // its parent instead of that node, so we'll update the misses:
+                let reroute_miss = |internal: &NodeOperator, miss: &mut Miss| {
```

Comment:
I think this has to recurse. Consider a join over a filtered projection. There are two layers of `query_through`.